### PR TITLE
Jl ccp snackbar

### PR DESCRIPTION
**Pull request by jesskeepswimming**
[Notion ticket](https://www.notion.so/uwblueprintexecs/CCP-patient-snackbar-link-039b5a16d08740d1a5bba2433876498e)

## Changes
- made the "view patient details" button in the ccp snackbar that pops up after transporting a patient functional
- added a new route dedicated to the patient details dialog in the top level App file
- added a PatientDetailsDialogRouting wrapper that allows external access of the dialog outside of the patient table in ccp
- added a redirect to the edit page on this function in case link is revisited 

## Screenshots
image
image

## Testing
- navigate to the ccp dashboard
(http://localhost:3000/events/3/ccps/1 should have patients)
- click on a patient that has not been transported yet
- edit patient status to transport
- complete patient transport editing and the snackbar should pop up
- click "view patient details" in the snackbar. the patient details dialog should pop up
- make sure all actions in the dialog work properly
- reload route. you should be redirected to the patient editing page

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [x] fill out `Changes`
  - [ ] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [x] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [x] make sure build passes
- [ ] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

File: src/components/PatientProfile/PatientProfilePage.tsx [hunk 1]
```diff
@@ -194,11 +194,17 @@ const PatientProfilePage = ({
...
+    <Link
+      to={{
+        pathname: `/events/${eventId}/ccps/${ccpId}/patientDetails/${patientId}`,
+        state: { modal: true },
+      }}
+      className="link"
+    >
       <Button style={{ color: Colours.SnackbarButtonBlue }}>
         View Patient Details
       </Button>
```
Comment (petorsong) [inline src/components/PatientProfile/PatientProfilePage.tsx]: is there a way to remove this `Button` tag entirely and replace it with the `Link` tag (and style it accordingly)

File: src/App.tsx [hunk 2]
```diff
@@ -93,6 +94,11 @@ function App() {
             path="/events/:eventId/ccps/:ccpId/patients/:patientId"
             component={(props) => <PatientProfilePage mode="edit" {...props} />}
           />
+
```
Comment (petorsong) [inline src/App.tsx]: nitpick
```suggestion
```

File: src/components/CCPDashboard/PatientDetailsDialog.tsx [hunk 3]
```diff
@@ -34,6 +50,75 @@ interface PatientDetail {
   styles?: object;
 }
 
+export const PatientDetailsDialogRouting = ({
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientDetailsDialog.tsx]: minor nit - is there a reason to export this at all?

File: src/components/CCPDashboard/PatientDetailsDialog.tsx [hunk 4]
```diff
@@ -34,6 +50,75 @@ interface PatientDetail {
...
+}) => {
+  const [openDetails, setOpenDetails] = React.useState(true);
+  const history = useHistory();
+  const classes = useStyles();
+
+  const { data } = useQuery(
+    patientId ? GET_PATIENT_BY_ID(patientId) : GET_ALL_PATIENTS
+  );
+
+  const patient: Patient = data ? data.patient : undefined;
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientDetailsDialog.tsx]: nit: usually makes more sense to set objects to `null` if you want it to be falsy
```suggestion
  const patient: Patient = data ? data.patient : null;
```

File: src/components/CCPDashboard/PatientDetailsDialog.tsx [hunk 5]
```diff
@@ -34,6 +50,75 @@ interface PatientDetail {
...
+              className={classes.editButton}
+            >
+              Edit
+            </Button>
+          </DialogActions>
+        </Dialog>
+      ) : (
+        <Redirect
+          to={`/events/${eventId}/ccps/${ccpId}/patients/${patientId}`}
+        />
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientDetailsDialog.tsx]: how does this redirect work? it only does this if `patient` is falsy (which imo implies that the `patientId` is invalid), and attempts to go to the edit form for that patient (which wouldn't work if the `patientId` is invalid)

**Review by petorsong — CHANGES_REQUESTED** [review]
nice first PR! 🎉 
couple of comments that need to be addressed from me

File: src/components/PatientProfile/PatientProfilePage.tsx (hunk 1 above)
Comment (jesskeepswimming) [reply src/components/PatientProfile/PatientProfilePage.tsx]: opted for cleaner onClick event

File: src/components/CCPDashboard/PatientDetailsDialog.tsx (hunk 3 above)
Comment (jesskeepswimming) [reply src/components/CCPDashboard/PatientDetailsDialog.tsx]: yes, import { PatientDetailsDialogRouting } from './components/CCPDashboard/PatientDetailsDialog'; is called in the App file

File: src/components/CCPDashboard/PatientDetailsDialog.tsx (hunk 5 above)
Comment (jesskeepswimming) [reply src/components/CCPDashboard/PatientDetailsDialog.tsx]: made it redirect to /events

File: src/components/CCPDashboard/PatientDetailsDialog.tsx [hunk 6]
```diff
@@ -34,6 +50,73 @@ interface PatientDetail {
...
+  const patient: Patient = data ? data.patient : null;
+
+  const handleCloseDetails = () => {
+    setOpenDetails(false);
+    history.push(`/events/${eventId}/ccps/${ccpId}`);
+  };
+
+  return (
+    <div>
+      {patient ? (
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientDetailsDialog.tsx]: can you explain why we need to have this check for `patient` and why we want to redirect to `/events` if it's null/falsy? i'm just unsure when `data.patient` will come out as null - can you describe a scenario when that might happen?

File: src/components/CCPDashboard/PatientDetailsDialog.tsx [hunk 7]
```diff
@@ -34,6 +50,73 @@ interface PatientDetail {
   styles?: object;
 }
 
+export const PatientDetailsDialogRouting = ({
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientDetailsDialog.tsx]:
```suggestion
export const PostTransportPatientDetailsDialog = ({
```

File: src/components/CCPDashboard/PatientDetailsDialog.tsx [hunk 8]
```diff
@@ -34,6 +50,73 @@ interface PatientDetail {
...
+      ccpId: string;
+    };
+  };
+}) => {
+  const [openDetails, setOpenDetails] = React.useState(true);
+  const history = useHistory();
+  const classes = useStyles();
+
+  const { data } = useQuery(
+    patientId ? GET_PATIENT_BY_ID(patientId) : GET_ALL_PATIENTS
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientDetailsDialog.tsx]: we don't need to fetch `GET_ALL_PATIENTS`

File: src/App.tsx [hunk 9]
```diff
@@ -93,6 +94,10 @@ function App() {
             path="/events/:eventId/ccps/:ccpId/patients/:patientId"
             component={(props) => <PatientProfilePage mode="edit" {...props} />}
           />
+          <Route
+            path="/events/:eventId/ccps/:ccpId/patientDetails/:patientId"
```
Comment (wongcmichelle) [inline src/App.tsx]: @petorsong I'm assuming this is the route that was decided upon?

**Review by wongcmichelle — CHANGES_REQUESTED** [review]
For consistency, I think we should refactor the `PatientDetailsDialog` to appear always in the same route (e.g. `/events/{eventId}/ccps/{ccpId}/{patientId}/view`). The route doesn't need to be different when we view from the snackbar after transporting vs. clicking on a row in the patient table. In short, I don't think we need the new `PatientDetailsDialogRouting` component. (Especially since the contents of the patient details dialog are no different in the 2 scenarios.)

@petorsong thoughts?

File: src/App.tsx (hunk 9 above)
Comment (petorsong) [reply src/App.tsx]: (just commenting here to address the PR comment as well)

i like the `.../view` route and the idea of not introducing a "duplicate" component if we can reuse our existing one

**Comment by jesskeepswimming** [comment]
@petorsong @wongcmichelle  I've made the requested changes, now the dialog opening will always be accompanied by a change in url. Testing procedure is the same as outlined originally. 

Changes:
- Attached all dialog instances with the route "/events/:eventId/ccps/:ccpId/open/:patientId". 
- "handleOpen" and "selectedPatient" states are now initialized based on whether there is a patientId parameter 
- the background is preserved
- closing the dialog redirects back to "/events/:eventId/ccps/:ccpId"

**Review by petorsong — APPROVED** [review]
holy crap this works super well, good job for getting this done! ✨ (and thanks @wongcmichelle for helping too!)

**Review by wongcmichelle — APPROVED** [review]
Looks great! Thank you 🚀

**Comment by wongcmichelle** [comment]
@petorsong just FYI this ticket doesn't include all of the other patient related snackbars (i think they were only recently added to Figma)

**Comment by petorsong** [comment]
> @petorsong just FYI this ticket doesn't include all of the other patient related snackbars (i think they were only recently added to Figma)

@wongcmichelle already accounted for - I've asked rishabh to audit our snackbars to see if we have all of them (this will go in a separate ticket)

**Pull request merged**